drivers/storage/portworx/component/autopilot.go

```diff
@@ -386,6 +387,64 @@ func (c *autopilot) getAutopilotDeploymentSpec(
 	maxSurge := intstr.FromInt(1)
 	imagePullPolicy := pxutil.ImagePullPolicy(cluster)
 
+	// If tls enabled, add env for autopilot and all apps using openstorage client:
+	// (see vendor/github.com/libopenstorage/openstorage/volume/drivers/pwx/connection.go)
+	// CaCertSecretEnv:             "PX_CA_CERT_SECRET",
+	// CaCertSecretKeyEnv:          "PX_CA_CERT_SECRET_KEY",
+	// assumption: user has already uploaded CA cert to the specified k8s secret
+	// Also set
+
+	if pxutil.IsTLSEnabledOnCluster(&cluster.Spec) {
+		// Set:
+		//	  env:
+		//    - name: PX_CA_CERT_SECRET
+		//      value: <default>
+		//    - name: PX_CA_CERT_SECRET_KEY
+		//      value: <default>
+		//    - name: PX_ENABLE_TLS
+		//	    value: "true"
+		CASecretNameValue := ""
+		CASecretKeyValue := ""
+		EnableTLSValue := ""
+		if cluster.Spec.Autopilot != nil && cluster.Spec.Autopilot.Env != nil {
+			CASecretNameValue = k8sutil.GetValueFromEnv(pxutil.EnvKeyAutopilotCASecretName, cluster.Spec.Autopilot.Env)
+			CASecretKeyValue = k8sutil.GetValueFromEnv(pxutil.EnvKeyAutopilotCASecretKey, cluster.Spec.Autopilot.Env)
+			EnableTLSValue = k8sutil.GetValueFromEnv(pxutil.EnvKeyPortworxEnableTLS, cluster.Spec.Autopilot.Env)
```

Comment:
When TLS is enabled in Portworx, the clients will have to use TLS, right? Or is that optional?
Assuming that it is necessary for clients to use TLS, we should not ask users to pass the environment variables twice in the outer env vars as well autopilot envs. In future in stork envs as well.

Reply:
Should we be passing the secret name and key to the clients or just mount the secret directly as a file in them?

Reply:
> Should we be passing the secret name and key to the clients or just mount the secret directly as a file in them?

That's a good idea. We will probably do that for the "server cert" and the "server key" that needs to be passed only to porx (i.e. mount the secret either as file or env)

For the CA cert, `openstorage`expects the name of the secret and doesn't read any environment variable
(https://github.com/libopenstorage/openstorage/blob/baba77246a2578575885654556cb538d2869fbbe/volume/drivers/pwx/connection.go#L73)
Therefore all application which need to connect to porx using tls (autopilot, stork) will need to be passed the secret name.

To make the change you suggest, we will need to change this here: https://github.com/libopenstorage/openstorage/blob/master/volume/drivers/pwx/connection.go#L178 ,
push the change to `openstorage`, update vendor packages in multiple repos etc etc.

Reply:
I don't have a strong preference over passing the values as secret names or files. Just thought that it would be easier to mount it as a file as most ssl libraries consume files, so the client/server does not have to talk to k8s every time to get the CA or server certs.

@puremudassir @lpabon On a different note, not directly related to this PR - I am assuming we are enabling TLS for Kubernetes only, at least initially. But reading from a k8s secret from openstorage means it would work only for k8s. I am not sure if we ever plan to support other orchestrators, but you never know. In such a case, would it be better to read the certs as a file?